engine/src/main/java/org/destinationsol/ResolutionProvider.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2018 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.destinationsol;
+
+import com.badlogic.gdx.Graphics;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+class ResolutionProvider {
+
+    private List<Resolution> resolutions;
+    private int currentPosition;
+
+    ResolutionProvider(List<Graphics.DisplayMode> displayModes) {
+        currentPosition = 0;
+        resolutions = displayModes.stream().map((displayMode) -> new Resolution(displayMode.width, displayMode.height))
+                .sorted(Comparator.comparing(Resolution::getWidth).thenComparing(Resolution::getHeight))
+                .collect(Collectors.toList());
+    }
+
+    public Resolution increase() {
+        return resolutions.get(++currentPosition % resolutions.size());
```

Comment:
This way this is implemented, I think it can cause a sneaky problem later on. Suppose there are 5 resolutions, someone calls increase() 5 times (leading to currentPosition having a final value 5), and then calls getResolution. What do we expect? 0th resolution. What do we get? OutOfBounds. I think we should split this line into
```currentPosition = (currentPosition + 1) % resolutions.size();
return resolutions.get(currentPosition);```
Let me know if I'm mistaken.

Reply:
:+1:

Reply:
Thanks! Done.